src/main/scala/com.snowplowanalytics.sauna/apis/UrbanAirship.scala

```diff
@@ -0,0 +1,166 @@
+package com.snowplowanalytics.sauna
+package apis
+
+
+
+import java.io.DataOutputStream
+import scala.concurrent.Future
+import scala.concurrent.ExecutionContext
+import scala.util.{ Failure, Success }
+import org.apache.commons.io.IOUtils
+import java.io.File
+import java.util.Properties
+import java.io.FileInputStream
+import java.io.FileNotFoundException
+import java.io.IOException
+import java.io.InputStream
+import java.util.ArrayList
+import java.io.InputStream
+import java.util.HashMap
+import java.util.Iterator
+import scala.io.Source.fromFile
+import java.net.URL
+import akka.actor.{ActorRef, Props}
+import java.util.Map.Entry
+import scala.io.Source
+import scala.collection
+import scala.collection.mutable.{ Map => MutableMap }
+import javax.net.ssl.HttpsURLConnection
+import java.util.zip.GZIPOutputStream
+import org.json.JSONObject
+import responders.urbanairship.UAResponder.Airship
+import scala.collection.immutable.ListMap
+import play.api.libs.json.Json
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import scala.concurrent.ExecutionContext.Implicits.global
+
+class UrbanAirship(implicit logger: ActorRef) {
+  import UrbanAirship._
+  
+//  implicit val ec = ExecutionContext.fromExecutor(Executors.newFixedThreadPool(10))
+  def MaptoRequest(appMap: Map[String, Map[String, List[Airship]]]): Unit =
+    {
+
+      var listNamesToKeyMap = Map[String, String]()
+      for ((appKey, value) <- appMap) {
+        var count = 0;
+        for ((listName, value1) <- value) {
+          val f = Future {
+
+            val url = urlPrefix + listName + "/csv/"
+            val obj: URL = new URL(url)
+            val con: HttpsURLConnection = obj.openConnection().asInstanceOf[HttpsURLConnection]
+
+            con.setRequestMethod("PUT")
+            
+             val urbanairshipFile = new File(Sauna.respondersLocation+"/urban_airship_config.json")
```

Comment:
This (and related actions) can be done outside `Future`. Try to enclose in `Future` as least lines of code as possible, especially if these actions can result in IO exceptions.